### PR TITLE
Fix #434, fix input to hash function in user_profile

### LINKED ISSUE #434: TypeError in djangoproject.com/~username/
Bug from the Python 3 migration:
```
TypeError: Unicode-objects must be encoded before hashing
  File "accounts/views.py", line 22, in user_profile
    'email_hash': hashlib.md5(u.email).hexdigest(),
```

### PR DESCRIPTION
**Pull request by relekang**

### CODE REVIEW

**Comment by timgraham** [comment]
Is there any chance to add a test? Even if it just ensures the view doesn't throw a 500.

**Comment by relekang** [comment]
I added a test just now.

File: accounts/views.py [hunk 1]
```diff
@@ -16,15 +16,14 @@
...
-        'email_hash': hashlib.md5(u.email).hexdigest(),
-        'user_can_commit': u.has_perm('auth.commit'),
-        'clas': find_agreements(u),
-        'stats': get_user_stats(u),
-    }
-    return render(request, "accounts/user_profile.html", ctx)
+    user = get_object_or_404(User, username=username)
+    return render(request, "accounts/user_profile.html", {
+        'user_obj': user,
+        'email_hash': hashlib.md5(user.email.encode('ascii', 'ignore')).hexdigest(),
```
Comment (timgraham) [inline accounts/views.py]: It might be nice to add a User suclass with a `get_email_hash()` method. This could be a separate change though.

Comment (relekang) [inline accounts/views.py]: I think it would be better as separate change, at least to get the bug fixed.

**Pull request merged**